State.h

```diff
@@ -0,0 +1,82 @@
+#ifndef STATE_H
+#define STATE_H
+
+#include "BreathData.h"
+
+/*
+ *  Parent class for all instantiated state classes
+ */
+class State {
+  public:
+    virtual State* update() = 0;           // perform maintenance, and perhaps transition to new state
+    static BreathData* breath;             // holds shared information about the breath cycle
+};
+
+/*
+ *  Specific states used by the ventilator control.
+ *  Only a singleton instance of each is created, accessible via the get() method.
+ */
+
+// Standby mode.  This is the initial state.
+class OffState : public State {
+  public:
+    static State* enter();               // performs actions necessary upon entering the state
+    State* update();                     // perform maintenance, and perhaps transition to next state
+    static State* begin(BreathData);     // returns the OffState instance with breath data link
```

Comment:
The name `begin` is confusing in two ways: 1) What is it beginning? and 2) it conflicts with the standard library's use of `begin` which returns an iterator pointing to the first element of a container.  How about `instance` or `singleton` instead?

Reply:
Good point; I forgot about the name conflict.  This function is supposed to be called to fire up the state machine and retrieve the first state.  I'll rename it `initState`.  Although maybe we don't even need it if I initialize breath somewhere else.  Will consider further.